python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -1031,55 +1017,23 @@ def _getIOTimeOut(self):
             _timeOut = 5. # 5sec hardcoded default
         return _timeOut
 
-    def _getWriteAccess(self):
-        """ tries to get write access to the object.
-        Raise LockingError (or so) on fail """
-        root = self._getRoot()
-        reg = root._getRegistry()
+    def _getSessionLock(self):
```

Comment:
This is up for discussion as far as I'm concerned as in the original code, `_getReadAccess` would do the load but **not** acquire the Session Lock but `_getWriteAccess` would try to acquire a Session Lock but load even if that failed. I would perfectly happy to link the two (i.e. grab the Session lock when we load an object) but that leaves the question of releasing the Session lock as at the moment, we can't 'unload' an object. Now, we may **not** want to ever release a Session Lock in which case that is a moot point 😄